modules/photo/src/seamless_cloning.cpp

```diff
@@ -72,36 +72,12 @@ void cv::seamlessClone(InputArray _src, InputArray _dst, InputArray _mask, Point
     Mat gray_inner = gray(Rect(1, 1, gray.cols - 2, gray.rows - 2));
     copyMakeBorder(gray_inner, gray, 1, 1, 1, 1, BORDER_ISOLATED | BORDER_CONSTANT, Scalar(0));
 
-    int minx = INT_MAX, miny = INT_MAX, maxx = INT_MIN, maxy = INT_MIN;
-    int h = gray.size().height;
-    int w = gray.size().width;
-
-    for(int i=0;i<h;i++)
-    {
-        for(int j=0;j<w;j++)
-        {
-            if(gray.at<uchar>(i,j) == 255)
-            {
-                miny = std::min(miny,i);
-                maxy = std::max(maxy,i);
-                minx = std::min(minx,j);
-                maxx = std::max(maxx,j);
-            }
-        }
-    }
-
-    int lenx = maxx - minx + 1;
-    int leny = maxy - miny + 1;
-
-    int minxd = p.x - lenx/2;
-    int minyd = p.y - leny/2;
-
-    Rect roi_d(minxd,minyd,lenx,leny);
-    Rect roi_s(minx,miny,lenx,leny);
+    Rect roi_s = boundingRect(gray);
+    Rect roi_d(p.x - roi_s.width / 2, p.y - roi_s.height / 2, roi_s.width, roi_s.height);
```

Comment:
is it better like
`Rect roi_d(p.x - roi_s.width * 0.5, p.y - roi_s.height * 0.5, roi_s.width, roi_s.height);`

Reply:
Rect is 'int'-based type. No need to use '0.5' here (also compiler should generate warning here due implicit cast `float -> int`).